recipes-bsp/grub/grub_%.bbappend

```diff
@@ -0,0 +1,35 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
+
+SRC_URI = " \
+    git://github.com/TrenchBoot/grub.git;branch=${BRANCH};protocol=https \
+    file://0001-add-root-flag-to-grub-bios-setup.patch \
+"
+
+# nooelint: oelint.append.protvars
+PV = "2.06"
+
+BRANCH = "intel-txt-aem-2.06-rebased"
+# nooelint: oelint.append.protvars
+SRCREV = "f6dfae51de0fb810bd441889c499db0602934db5"
+
+S = "${WORKDIR}/git"
+
+FILES:${PN}-common += " \
+    ${libdir}/grub-tb \
+    ${libdir}/grub/i386-pc \
+"
+
+RDEPENDS:${PN}-common += "diffutils freetype"
+
+do_configure:prepend() {
+    cd ${S}
+    ${S}/bootstrap
+    ${S}/autogen.sh
+}
+
+do_install:append () {
+    install -d ${D}${libdir}/grub-tb
+    cp -r ${D}${libdir}/grub/i386-pc ${D}${libdir}/grub-tb
```

Comment:
If we switched to append, is there any reason why need this `grub-tb` directory?
How is it different in terms of content from the regular grub directory? Is there any difference at all?

Reply:
@macpijan 
It's used in wic plugin `bootimg-grub-tb.py`

```python
# Copying grub modules
        grub_dir_native = get_bitbake_var("IMAGE_ROOTFS") + "/usr/lib/grub-tb"
        shutil.copytree("%s/i386-pc" % (grub_dir_native),
                        "%s/grub/i386-pc" % hdddir)
```

Removing this directory from bbappend and plugin didn't work 

```
| ERROR: _exec_cmd: grub-mkimage -p (hd0,msdos1)/grub -d /build/tmp/work/genericx86_64-tb-linux/tb-minimal-image/1.0/tmp-wic/hdd/boot/grub/i386-pc -o /build/tmp/work/genericx86_64-tb-linux/tb-minimal-image/1.0/tmp-wic/hdd/boot/grub/i386-pc/core.img -O i386-pc at_keyboard biosdisk boot chain configfile ext2 fat linux ls part_msdos reboot serial vga returned '1' instead of 0
| output: grub-mkimage: error: cannot open `/build/tmp/work/genericx86_64-tb-linux/tb-minimal-image/1.0/tmp-wic/hdd/boot/grub/i386-pc/core.img': No such file or directory.
```

So removing this directory would probably need more changes if it's possible at all (I didn't create this plugin so I'm not exactly sure how it works)

Reply:
Removed in: [6b67c32](https://github.com/3mdeb/meta-trenchboot/pull/28/commits/6b67c3296d5733331022c989950f9e0a3b2f2b6c)